src/silicone/utils.py

```diff
@@ -526,3 +526,38 @@ def _construct_consistent_values(aggregate_name, components, db_to_generate):
     use["unit"] = units.iloc[0]
     use["variable"] = aggregate_name
     return pyam.IamDataFrame(use)
+
+
+def _make_weighting_series(df, weights):
+    """
+    Makes a complete list of weights for all scenarios from a dictionary of only
+    unusual weights.
+    Parameters
+    ----------
+    df: pd.DataFrame
+        The timseries with the full set of models and scenarios whose weights
+        should be returned
+    weights: Dict{(str, str) : float}
+        The dictionary, mapping the (model and scenario) tuple onto the weight (relative
+        to a weight of 1 for the default). This does not have to include all scenarios
+        in df, but cannot include scenarios not in df.
+
+    Returns
+    -------
+    Series
+        A series with index corresponding to the timeseries of the dataframe and values
+        corresponding to the weights.
+
+    """
+    all_mod_scen = [(i[0], i[1]) for i in df.index]
```

Comment:
```suggestion
    all_mod_scen = [(i[0], i[1]) for i in df.index.drop_duplicates()]
```

Something like that might remove the need for the set call below?

Reply:
ping

Reply:
That would be a lot neater!